docs/large_inputs.md

```diff
@@ -1,63 +1,127 @@
-may set `--num_workers <num workers>` to use the exact
-number of workers.
+## Important note about adjusting quota
 
-### Adjusting quota
+Compute Engine enforces quota on maximum amount of resources that can be used
+at any time for variety of reasons. Please see
+https://cloud.google.com/compute/quotas for more details. As a result, you may
+need to adjust the quota to satisfy the job requirements.
 
-To increase the number of workers, you need to adjust the Compute Engine quota.
-In particular, you need at least `num_workers + 1` IPs, `num_workers + 1` CPUs,
-and `num_workers * 250GB` of disk in the zone you are running the pipeline.
-Please see https://cloud.google.com/compute/quotas for details on how to
-adjust your quota.
+The main Compute Engine quotas to be adjusted are:
+* `In-use IP addresses`: One per worker.
+* `CPUs`: At least one per worker. More if larger machine type is used.
+* `Persistent Disk Standard (GB)`: At least 250GB per worker. More if larger
+  disk is used.
+* `Persistent Disk SSD (GB)`: Only needed if `--worker_disk_type` is set to SSD.
+  Required quota size is the same as `Persistent Disk Standard`.
 
-Note: the `+1` in the above figures is to account for the additional VM used as
-the "pipeline orchastrator" when using the
-[pipelines API with the docker image](../README.md#using-docker).
+## `--optimize_for_large_inputs`
 
-## Machine type
+This flag should be set to true when loading more than 50,000 files and/or
+[merging](variant_merging.md) is enabled for a large number (>3 billion)
+of variants. This flag optimizes the Dataflow pipeline for large inputs, which
+can save significant cost/time, but the additional overhead may hurt cost/time
+for small inputs.
+
+## `--max_num_workers`
+
+By default, Dataflow uses its autoscaling algorithm to adjust the number of
+workers assigned to each job. You may adjust the maximum number of workers using
+`--max_num_workers`. You may also use `--num_workers` to specify the initial
+number of workers to assign to the job.
+
+## `--worker_machine_type`
 
 By default, Dataflow uses the `n1-standard-1` machine, which has 1 vCPU and
-3.75GB of RAM. It is recommanded to use a larger machine type for large
-datasets. You may change the machine type by adding
-`--worker_machine_type <machine type>` to the pipeline args. Please see
-https://cloud.google.com/compute/pricing#predefined_machine_types for a list of
-available machine types.
+3.75GB of RAM. You may need to request a larger machine for large datasets.
+Please see https://cloud.google.com/compute/pricing#predefined_machine_types
+for a list of available machine types.
+
+We have observed that Dataflow performs more efficiently when running
+with a large number of small machines rather than a small number of large
+machines (e.g. 200 `n1-standard-4` workers instead of 25 `n1-standard-32`
+workers). This is due to disk/network IOPS (input/output operations per second)
```

Comment:
(e.g. having `CPUs` quota set to 800, 200 `n1-standard-4` workers instead of 25 `n1-standard-32` workers)

Reply:
I do like to avoid coupling this point with quota here; as this can be independent of the quota (project can still have more CPUs but user may want to just use 800 CPUs).